rpc/types/src/misc/distribution.rs

```diff
@@ -0,0 +1,298 @@
+//! Output distributions for [`crate::json::GetOutputDistributionResponse`].
+
+//---------------------------------------------------------------------------------------------------- Use
+use std::mem::size_of;
+
+#[cfg(feature = "serde")]
+use crate::serde::{serde_false, serde_true};
+use cuprate_epee_encoding::read_varint;
+#[cfg(feature = "serde")]
+use serde::{Deserialize, Serialize};
+
+#[cfg(feature = "epee")]
+use cuprate_epee_encoding::{
+    epee_object, error,
+    macros::bytes::{Buf, BufMut},
+    read_epee_value, write_field, write_varint, EpeeObject, EpeeObjectBuilder, EpeeValue, Marker,
+};
+
+use super::OutputDistributionData;
+
+//---------------------------------------------------------------------------------------------------- Free
+/// Used for [`Distribution::CompressedBinary::compressed_data`].
+///
+/// TODO: for handler code. This should already be provided in the field.
+#[doc = crate::macros::monero_definition_link!(
+    cc73fe71162d564ffda8e549b79a350bca53c454,
+    "rpc/core_rpc_server_commands_defs.h",
+    45..=55
+)]
+#[allow(clippy::needless_pass_by_value)] // TODO: remove after impl
+fn compress_integer_array(array: Vec<u64>) -> error::Result<Vec<u64>> {
+    todo!()
+}
+
+/// Used for [`Distribution::CompressedBinary::compressed_data`].
+///
+/// TODO: for handler code. This should already be provided in the field.
+#[doc = crate::macros::monero_definition_link!(
+    cc73fe71162d564ffda8e549b79a350bca53c454,
+    "rpc/core_rpc_server_commands_defs.h",
+    57..=72
+)]
+#[allow(clippy::needless_pass_by_value)] // TODO: remove after impl
+fn decompress_integer_array(array: Vec<u64>) -> Vec<u64> {
+    todo!()
+}
```

Comment:
`Distribution::CompressedBinary` has a `compressed_data` field that gets populated _during_ serialization: https://github.com/monero-project/monero/blob/cc73fe71162d564ffda8e549b79a350bca53c454/src/rpc/core_rpc_server_commands_defs.h#L2487.

I think this is a bit weird since this is a field in a data structure, it should be provided by the constructor themselves - i.e. I think the handler should handle this, otherwise we have to comment something like:

```markdown
# Invariant
Leave the `compressed_data` field empty, the serialization will handle it...?
```

Reply:
For the JSON version of this type we will need a custom function that writes this type out right?

I think you could use the `serde` from/into attributes to go between a type that has all fields and the type we expose to users.

However if you would rather the handler handle this then I am ok with that

Reply:
> For the JSON version of this type we will need a custom function that writes this type out right?

Probably yes, but because of the binary string, not the compression.

There is no JSON compressed type, since it must be `binary && compress`: https://github.com/monero-project/monero/blob/cc73fe71162d564ffda8e549b79a350bca53c454/src/rpc/core_rpc_server_commands_defs.h#L2481-L2504.

> However if you would rather the handler handle this then I am ok with that

Or maybe we could hide the (de)compression in the (de)serialization? I.e. there is no `compressed_data` field, only `distribution`. It is always uncompressed from the caller's perspective because the epee (de)serialization handles it internally.

Reply:
> Or maybe we could hide the (de)compression in the (de)serialization?

Yeah this is what I was getting at, up to you what you think would be better